cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaConnectConfigurationBuilderTest.java

```diff
@@ -514,4 +530,202 @@ public void withPluginPath() {
+                                + "org.apache.kafka.common.metrics.JmxReporter\n"
+                                + "producer.metric.reporters="
+                                + "org.apache.kafka.common.metrics.JmxReporter\n"
+                                + "consumer.metric.reporters="
+                                + "org.apache.kafka.common.metrics.JmxReporter"
+                ),
+                Arguments.of(userConfig, false, true,
+                        expectedConfig
+                                + "metric.reporters="
+                                + "my.domain.CustomMetricReporter,"
+                                + StrimziMetricsReporterConfig.KAFKA_CLASS + "\n"
+                                + "admin.metric.reporters="
+                                + StrimziMetricsReporterConfig.KAFKA_CLASS + "\n"
+                                + "producer.metric.reporters="
+                                + StrimziMetricsReporterConfig.KAFKA_CLASS + "\n"
+                                + "consumer.metric.reporters="
+                                + StrimziMetricsReporterConfig.KAFKA_CLASS + "\n"
+                ),
+
+                Arguments.of(userConfig, true, true,
+                        expectedConfig
+                                + "metric.reporters="
+                                + "my.domain.CustomMetricReporter,"
+                                + "org.apache.kafka.common.metrics.JmxReporter,"
+                                + StrimziMetricsReporterConfig.KAFKA_CLASS + "\n"
+                                + "admin.metric.reporters="
+                                + "org.apache.kafka.common.metrics.JmxReporter,"
+                                + StrimziMetricsReporterConfig.KAFKA_CLASS + "\n"
+                                + "producer.metric.reporters="
+                                + "org.apache.kafka.common.metrics.JmxReporter,"
+                                + StrimziMetricsReporterConfig.KAFKA_CLASS + "\n"
+                                + "consumer.metric.reporters="
+                                + "org.apache.kafka.common.metrics.JmxReporter,"
+                                + StrimziMetricsReporterConfig.KAFKA_CLASS + "\n"
+                ));
+    }
+
+    @ParameterizedTest
+    @MethodSource("sourceUserConfigWithMetricsReporters")
+    public void testUserConfigurationWithMetricReporters(
+            KafkaConnectConfiguration userConfig,
+            boolean injectJmx,
+            boolean injectStrimzi,
+            String expectedConfig) {
+        String actualConfig = new KafkaConnectConfigurationBuilder(Reconciliation.DUMMY_RECONCILIATION, BOOTSTRAP_SERVERS)
+                .withUserConfiguration(userConfig, injectJmx, injectStrimzi)
+                .build();
+        assertThat(actualConfig, isEquivalent(expectedConfig));
+    }
+
+    @ParallelTest
+    public void testStrimziMetricsReporterViaUserAndMetricsConfigs() {
```

Comment:
Wouldn't it make sense to add it to the parametrized test above? It seems that would capture the core of the test.

Reply:
This is a corner case, and I would like to keep it separated.

Reply:
Ok, good.